### PR TITLE
JAX-RS Spec: Stop generating @NotNull Validation for required & optional properties

### PR DESCRIPTION
**Pull request by wiesnery**
Hi there,

I just realized a bug / missing feature in the JAX-RS Spec server stub generator. When providing a property which is required and readonly, it should according to the specification not be validated on request time. [spec](https://swagger.io/specification/#schema-object)
I used this combination for an id field, which is obviously not present upon posting a new entity but has to be present for reference in all further responses from the server.

I just adapted the code from the spring generator, as it works as expected: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaSpring/beanValidation.mustache

You can find a similar issue with the Vert.X generator here: https://groups.google.com/g/vertx/c/CS5ZPIF8RgU?pli=1

Here you can find the results after the change (left side, the property is not required, : 
image

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### CODE REVIEW

**Comment by wiesnery** [comment]
Just seeing, that I might need to rebase. I branched off from master and thought that 6.0.X should be ahead of it but it seems not to be the case.

Please instruct which branch to rebase on.

File: samples/server/petstore/jaxrs-spec-required-and-readonly-property/src/gen/java/org/openapitools/model/ReadonlyAndRequiredProperties.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  /**
+   **/
+  public ReadonlyAndRequiredProperties requiredYesReadonlyYes(String requiredYesReadonlyYes) {
+    this.requiredYesReadonlyYes = requiredYesReadonlyYes;
+    return this;
+  }
+
+  
+  @ApiModelProperty(required = true, value = "")
+  @JsonProperty("requiredYesReadonlyYes")
```
Comment (wiesnery) [inline samples/server/petstore/jaxrs-spec-required-and-readonly-property/src/gen/java/org/openapitools/model/ReadonlyAndRequiredProperties.java:42]: The change is now, that the @NotNull is not present anymore when generating the server stub for a property which is marked as readonly and required

**Comment by wiesnery** [comment]
@jorgerod @wing328 could you have a look at my PR? No activity since 1 week and you guys seem to be active recently.

**Comment by wiesnery** [comment]
@jimschubert 3 weeks have passed now - could you please comment, review or redirect?
Its really disappointing to contribute here...
@wing328 I have seen you were active recently. Maybe you know the right person to assign this PR to?

**Comment by wing328** [comment]
Sorry for the delay in reviewing this PR.

LGTM

**Pull request merged**